File: output/CodeGen/codenet/C++/Java/s612033342.java
public class s612033342 {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    int h = in.nextInt();
    int w = in.nextInt();
    int k = in.nextInt();
    int[][] hw = new int[h][w];
    for (int i = 0; i < h; i++) {
      for (int j = 0; j < w; j++) {
        hw[i][j] = in.nextInt();
      }
    }
    int sum = 0;
    for (int i = 0; i < pow2(h); i++) {
      for (int j = 0; j < pow2(w); j++) {
        if (count(hw, i, j) == k) sum++;
      }
    }
    System.out.println(sum);
  }
}

File: output/CodeGen/codenet/C++/Python/s612033342.py
def count(hw, a, b):
  sum = 0
  for i in range(len(hw)):
    for j in range(len(hw[0])):
      if a >> i & 1 and b >> j & 1 and hw[i][j] == 1:
        sum += 1
  return sum

def pow2(a):
  ret = 1
  for i in range(a):
    ret *= 2
  return ret

def main():
  h, w, k = map(int, raw_input().split())
  hw = [[0 for _ in range(w)] for _ in range(h)]
  for i in range(h):
    for j in range(w):
      hw[i][j] = raw_input() == '#'
  sum = 0
  for i in range(pow2(h)):
    for j in range(pow2(w)):
      if count(hw, i, j) == k:
        sum += 1
  print sum

if __name__ == '__main__':
  main()